FunctionExecutor fails when ChatGPT returns "" as function call parameters.

ChatGPT occasionally returns empty string as function call arguments parameter when function does not have any arguments.

e.g.
```
{
  "role": "assistant",
  "tool_calls": [
    {
      "id": "some_id",
      "type": "function",
      "function": {
        "name": "some_function",
        "arguments": ""
      }
    }
  ]
}      
```

and FunctionExecutor throws with
```
Caused by: io.github.sashirestela.cleverclient.support.CleverClientException: Cannot convert the Json  to class com.invisible.framework.hmaa.web.automation.instruction.ExecuteWithoutInstructionTool.
	at io.github.sashirestela.cleverclient.util.JsonUtil.jsonToObject(JsonUtil.java:48)
	at io.github.sashirestela.openai.common.function.FunctionExecutor.execute(FunctionExecutor.java:81)
	... 42 more
Caused by: com.fasterxml.jackson.databind.exc.MismatchedInputException: No content to map due to end-of-input
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1]
	at com.fasterxml.jackson.databind.exc.MismatchedInputException.from(MismatchedInputException.java:59)
	at com.fasterxml.jackson.databind.DeserializationContext.reportInputMismatch(DeserializationContext.java:1767)
	at com.fasterxml.jackson.databind.ObjectReader._initForReading(ObjectReader.java:360)
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:2115)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1566)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1586)
	at io.github.sashirestela.cleverclient.util.JsonUtil.jsonToObject(JsonUtil.java:46)
	... 43 more
```

This is probably a bad response from OpenAI, but would be nice we can gracefully handle this case.

--- a/src/main/java/io/github/sashirestela/openai/common/function/FunctionExecutor.java
+++ b/src/main/java/io/github/sashirestela/openai/common/function/FunctionExecutor.java
@@ -78,7 +78,9 @@ public class FunctionExecutor {
         }
         try {
             var function = mapFunctions.get(functionName);
-            var object = JsonUtil.jsonToObject(functionCall.getArguments(), function.getFunctionalClass());
+            var object = JsonUtil.jsonToObject(
+                    functionCall.getArguments().isBlank() ? "{}" : functionCall.getArguments(),
+                    function.getFunctionalClass());
             return (T) object.execute();
         } catch (RuntimeException e) {
             throw new SimpleUncheckedException("Cannot execute the function {0}.", functionName, e);

--- a/src/test/java/io/github/sashirestela/openai/common/function/FunctionExecutorTest.java
+++ b/src/test/java/io/github/sashirestela/openai/common/function/FunctionExecutorTest.java
@@ -129,8 +129,8 @@ class FunctionExecutorTest {
     void shouldReturnListOfFunctionsWhenToolChoiceIsPassed() {
         Object[][] testData = {
                 { ToolChoiceOption.NONE, 0 },
-                { ToolChoiceOption.AUTO, 2, "convert_to_celsius", "exponentiation" },
-                { ToolChoiceOption.REQUIRED, 2, "convert_to_celsius", "exponentiation" },
+                { ToolChoiceOption.AUTO, 3, "convert_to_celsius", "exponentiation", "get_random_number" },
+                { ToolChoiceOption.REQUIRED, 3, "convert_to_celsius", "exponentiation", "get_random_number" },
                 { ToolChoice.function("exponentiation"), 1, "exponentiation" }
         };
         var executor = new FunctionExecutor(functionList);

--- a/src/test/java/io/github/sashirestela/openai/common/function/FunctionExecutorTest.java
+++ b/src/test/java/io/github/sashirestela/openai/common/function/FunctionExecutorTest.java
@@ -26,6 +26,11 @@ class FunctionExecutorTest {
                     .name("exponentiation")
                     .description("Multiply a base number a number n times")
                     .functionalClass(MathPower.class)
+                    .build(),
+            FunctionDef.builder()
+                    .name("get_random_number")
+                    .description("Get random number")
+                    .functionalClass(RandomNumber.class)
                     .build());
 
     @Test
@@ -155,6 +160,15 @@ class FunctionExecutorTest {
         }
     }
 
+    @Test
+    void shouldHandleBlankArgumentsStringGracefully() {
+        var executor = new FunctionExecutor(functionList);
+        var functionToCall = new FunctionCall("get_random_number", "");
+        var actualResult = executor.execute(functionToCall);
+        var expectedResult = 42;
+        assertEquals(actualResult, expectedResult);
+    }
+
     private void sortListFunction(List<Tool> list) {
         list.sort((o1, o2) -> o1.getFunction().getName().compareTo(o2.getFunction().getName()));
     }
@@ -178,6 +192,15 @@ class FunctionExecutorTest {
         @Override
         public Object execute() {
             return Math.pow(base, exponent);
+        }
+
+    }
+
+    static class RandomNumber implements Functional {
+
+        @Override
+        public Object execute() {
+            return 42;
         }
 
     }

